mmpy_bot/function.py

```diff
@@ -19,24 +19,32 @@
 class Function(ABC):
     def __init__(
         self,
-        function: Callable,
+        function: Union[Callable, click.command],
         matcher: re.Pattern,
         **metadata,
     ):
         # If another Function was passed, keep track of all these siblings.
         # We later use them to register not only the outermost Function, but also any
         # stacked ones.
         self.siblings = []
+
         while isinstance(function, Function):
             self.siblings.append(function)
             function = function.function
 
+        # FIXME: After this while loop it is possible that function is not a Function, do we really want to assign self.function to something which is not a Function? Check if this is needed for the click.Command case
```

Comment:
I think if this happens we would be in an error state.
It may be easier to simply raise an exception about improper use or improper state.
Could be a `raise ValueError(...)` if `self.function` is `None`.

Reply:
Oops I missed your response before merging. I'll add a new PR to address this FIXME.